Framework/MDAlgorithms/test/IntegratePeaksMD2Test.h

```diff
@@ -180,6 +180,128 @@ class IntegratePeaksMD2Test : public CxxTest::TestSuite {
+    Instrument_sptr inst = ComponentCreationHelper::createTestInstrumentCylindrical(5);
+    peakWS->addPeak(Peak(inst, 1, 1.0, V3D(0., 0., 0.)));
+    TS_ASSERT_EQUALS(peakWS->getPeak(0).getIntensity(), 0.0);
+    AnalysisDataService::Instance().addOrReplace("IntegratePeaksMD2Test_peaks", peakWS);
+
+    std::vector<double> peakRadius(numberValuesPeakRadius, 1.0);
+    std::vector<double> bkgInnerRadius(numberValuesBkgInnerRadius, 1.1);
+    std::vector<double> bkgOuterRadius(numberValuesBkgOuterRadius, 2.0);
+
+    IntegratePeaksMD2 alg;
+    TS_ASSERT_THROWS_NOTHING(alg.initialize())
+    TS_ASSERT(alg.isInitialized())
+    TS_ASSERT_THROWS_NOTHING(alg.setPropertyValue("InputWorkspace", "IntegratePeaksMD2Test_MDEWS"));
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("PeakRadius", peakRadius));
+    TS_ASSERT_THROWS_NOTHING(alg.setPropertyValue("PeaksWorkspace", "IntegratePeaksMD2Test_peaks"));
+    TS_ASSERT_THROWS_NOTHING(alg.setPropertyValue("OutputWorkspace", "IntegratePeaksMD2Test_Output"));
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("BackgroundInnerRadius", bkgInnerRadius));
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("BackgroundOuterRadius", bkgOuterRadius));
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("Ellipsoid", ellip));
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("Cylinder", cyl));
+    if (shouldPass) {
+      TS_ASSERT_THROWS_NOTHING(alg.execute());
+    } else {
+      TS_ASSERT_THROWS(alg.execute(), const std::runtime_error &);
+    }
+    AnalysisDataService::Instance().remove("IntegratePeaksMD2Test_peaks");
+  }
+  /** In the following validation tests, three validation principles must be met:
+      (where radius can be peak/BackgroundInner/BackgroundOuter)
+   A) radius vector has 1 or 3 values
+   B) radius vector has 1 value for sphere/cylinder/ellipsoid, 3 values only for ellipsoid
+      (sphere implied when ellipsoid=false and cylinder=false)
+   C) ellipsoid and cylinder cannot both be true
+   */
+  /// Peak Radius
+  void test_validationPeak1ValueSphere() { simpleRun(true, 1, 1, 1, false, false); }
+  void test_validationPeak1ValueCylinder() { simpleRun(true, 1, 1, 1, true, false); }
+  void test_validationPeak1ValueEllipsoid() { simpleRun(true, 1, 3, 3, false, true); }
+  void test_validationPeak1ValueEllipsoidAndCylinder() { simpleRun(false, 1, 1, 1, true, true); } // fail C
+
+  void test_validationPeak2ValuesSphere() { simpleRun(false, 2, 1, 1, false, false); }             // fail A
+  void test_validationPeak2ValuesCylinder() { simpleRun(false, 2, 1, 1, true, false); }            // fail A
+  void test_validationPeak2ValuesEllipsoid() { simpleRun(false, 2, 3, 3, false, true); }           // fail A
+  void test_validationPeak2ValuesEllipsoidAndCylinder() { simpleRun(false, 2, 1, 1, true, true); } // fail A + C
+
+  void test_validationPeak3ValuesSphere() { simpleRun(false, 3, 1, 1, false, false); }  // fail B
+  void test_validationPeak3ValuesCylinder() { simpleRun(false, 3, 1, 1, true, false); } // fail B
```

Comment:
I might have missed it but do we have a similar test to this for ellipsoid with `numberValuesPeakRadius=3`, `numberValuesBkgInnerRadius=1` and `numberValuesBkgOuterRadius=1`?

Reply:
No I don't seem to have 3-1-1 for ellipsoid. I'll have a think about how to test this. I also don't like all the duplicate tests I've commented out. I was trying to give all the "other radii" (the ones I'm not testing) a valid entry.